### ISSUE TITLE
Create index.xhtml files for non-core doc trees

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#10853]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)| @puzzlement reported|
|-|-|
|Trac ID|trac#1253|
|Type|defect|
|Created|2005-10-08 11:01:02Z|
|Branch|https://github.com/twisted/twisted/tree/non-core-docs-1253|

Attachments:

* [non-core-docs-1253.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/2c6/2c61422714bcb5935215adf442a8f5b1d80ffc54/402eb385d6187df2a1ebda0184d4f3f8d5e99031.patch) (6755 bytes) - added by collab on 2008-07-01 03:45:42Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__1253 1253
type__defect defect
reporter__hypatia hypatia
priority__high high
milestone__ 
branch__branches_non_core_docs_1253 branches/non-core-docs-1253
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1128769262000000 1128769262000000
changetime__1217372421000000 1217372421000000
version__ 
owner__ 
cc__thijs cc__radix cc__exarkun
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement commented|
|-|-|

```
#!html
<pre>
doc/core/index.xhtml exists, doc/core/[lore,flow,mail,news...]/index.xhtml doesn't.
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="puzzlement's avatar" src="https://avatars.githubusercontent.com/u/94942?s=50" width="50" height="50">](https://github.com/puzzlement)|@puzzlement set owner to edsuom|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|collab set owner to collab<br>collab set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|collab commented|
|-|-|

Attached a [[non-core-docs-1253.patch](None/non-core-docs-1253.patch) patch] that fixes this ticket.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner<br>@exarkun set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun commented|
|-|-|

Hmm.  This could be a step in the right direction.  There are two possibilities:

  1. the index files should be hand-written and maintained by hand
  1. the index files should be automatically generated

I'm fine with the first option, although I'd prefer the second, but even for the first option there's a bit more to be done.  These `.xhtml` files are inputs to lore and they need to be processed by the documentation generator.  Except for `core/index.xhtml`, these are all new files, so they're in directories no one was looking for documentation inputs in before.  This probably isn't really a big deal, since I think we still manually generate that directory list and feed it to the tool (but the plan is to not do that forever).  However, there's also the question of where the outputs will show up on the website.  I suppose in places like `http://twistedmatrix.com/projects/lore/documentation/`.  This involves changes to the website update process, which is also still manual.

So hm.  What conclusion did I reach?

The patch may be correct, but a couple other procedures need to be updated.  ReleaseAutomation and ReleaseProcedure might be related.  ReleaseProcedure already seems to be outdated wrt the current reality, though.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set owner to collab|
|-|-|

Oh, one more thing - the patch includes a change to doc/flow/index.xhtml, but there is no such file in the repository.  What is this a diff from/to?

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|collab commented|
|-|-|

Replying to [exarkun](#note_7):
> Oh, one more thing - the patch includes a change to doc/flow/index.xhtml, but there is no such file in the repository.  What is this a diff from/to?

Yeah, I can see how that one's confusing. I actually did a svn copy of the ```doc/core.index.html``` for that one, that's why those edits show up. The other new ones I added after that were normal copies.

**Comment by twisted-trac** [comment]
|<img alt="collab's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|collab removed owner|
|-|-|

Replying to [exarkun](#note_6):
> The patch may be correct, but a couple other procedures need to be updated.  ReleaseAutomation and ReleaseProcedure might be related.  ReleaseProcedure already seems to be outdated wrt the current reality, though.

I wouldn't mind helping out with that but I don't have sufficient permissions to edit those page.. Those also might require their own ticket, not sure. Adding it to the  review queue again, not sure how to improve the patch etc at this point.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_11"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Can you put this into a branch and run the documentation builder against it?  I've also given you permission to edit those wiki pages, in case you can think of any useful changes to make.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra set status to `assigned`|
|-|-|

I created a new [branches/non-core-docs-1253 branch](branches/non-core-docs-1253 branch). I'll figure out how to use the documentation builder. I also manually set the 'Branch' ticket field, let me know if that was a bad idea.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_13"></a>|@exarkun commented|
|-|-|

Setting the branch field manually isn't bad per se, however it should happen automatically when you create a branch for a ticket.  The fact that it didn't may have something to do with the fact that the branch for this ticket seems not to have been created in the usual way.  You should use Combinator, or mimic the way Combinator makes branches.  As the branch currently is, it's very difficult to merge (and therefore difficult to review).

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra commented|
|-|-|

Replying to [exarkun](#note_13):
> You should use Combinator, or mimic the way Combinator makes branches.  As the branch currently is, it's very difficult to merge (and therefore difficult to review).

Alright, sorry about that, will re-create the branch with Combinator.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_15"></a>|@thijstriemstra commented|
|-|-|

(In [24421]) Branching to 'non-core-docs-1253'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_16"></a>|@thijstriemstra removed owner<br>@thijstriemstra set status to `new`|
|-|-|

Replying to [exarkun](#note_13):
> As the branch currently is, it's very difficult to merge (and therefore difficult to review).

I created a new branch inc the patch using Combinator, putting it up for review.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_17"></a>|@exarkun set owner to @thijstriemstra|
|-|-|

Looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_18"></a>|@thijstriemstra set status to `closed`|
|-|-|

(In [24447]) Merge non-core-docs-1253: Created index.xhtml files for non-core doc trees

Author: thijs
Reviewer: exarkun
Fixes: [#10853](https://github.com/twisted/twisted/issues/10853)

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
